sycl/include/sycl/detail/vector_convert.hpp

```diff
@@ -930,7 +933,11 @@ vec<convertT, NumElements> vec<DataT, NumElements>::convert() const {
       auto val =
           detail::convertImpl<T, R, roundingMode, NumElements, OpenCLVecT,
                               OpenCLVecR>(bit_cast<OpenCLVecT>(*this));
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+      sycl::detail::memcpy_no_adl(&Result.m_Data, &val, sizeof(Result));
```

Comment:
Would there be a problem in just switching to `memcpy_no_adl` for all paths?

Reply:
We can, but I don't see any improvement with that. If we were to do that, we would have something like the following which doesn't simplify the code.
```
#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
      sycl::detail::memcpy_no_adl(&Result.m_Data, &val, sizeof(Result)); // For C-array
#else
     sycl::detail::memcpy_no_adl(Result.m_Data.data(), &val, sizeof(Result)); // For std::array
#endif
```
We couldn't have done `sycl::detail::memcpy_no_adl(&Result.m_Data, &val, sizeof(Result));` for `std::array` as base address of `std::array` is not guaranteed to be same as that of underlying C-array: https://stackoverflow.com/questions/69500721/is-the-address-of-a-stdarray-guaranteed-the-same-as-its-data

Reply:
On a second though, as SO answer mentioned, most popular implementations does guarantee `std::array` and underlying C-array to have same address. So, I can do something like:
```
static_assert(sizeof(Result.m_Data) == sizeof(DataT[AdjustedNum]))
sycl::detail::memcpy_no_adl(&Result.m_Data, &val, sizeof(Result)); // for both C-array and std::array
```

Reply:
I think having the `static_assert` should be fine. We only risk this hitting weird implementations until preview changes are promoted, so I think the risk is low while the reduced complexity is nice. 😄